src/cli/cli_parser.py

```diff
@@ -0,0 +1,95 @@
+from argparse import ArgumentParser, Namespace
+
+from src.cli.types import DPRPVerPair, Settings
+from src.definitions import PackFormatsDict, DiscListEntryContents, DiscListContents
+
+from src.cli.logger import *
+from csv import reader
+
+from pathlib import Path
+
+
+def is_legacy_version(mc_ver: str) -> bool:
+    mc_ver_val: list[int] = [int(i) for i in mc_ver.split(".")]
+    return mc_ver_val <= [1, 19, 3]
+
+
+def get_dp_rp_version(mc_ver: str) -> DPRPVerPair:
+    # find the corresponding dp&rp version of the game.
+    matching_version = None
+    for version, values in PackFormatsDict.items():
+        if '-' in version:
+            start, end = version.split(' - ')
+            if start <= mc_ver <= end:
+                matching_version = version
+                break
+        elif version == mc_ver:
+            matching_version = version
+            break
+
+    if matching_version:
+        info(f"matched game version:{matching_version}")
+        dp_rp_versions = PackFormatsDict[matching_version]
+        return dp_rp_versions
+    else:
+        error(f"unknown game version{mc_ver}。")
+
+
+def absolute_path(input_path: str):
+    return str(Path(input_path).absolute())
+
+
+class CLIParser:
+    _parse_result: Namespace
+
+    # Namespace(disc_entries_file='123', minecraft_version='1.2.0', use_concurrent=False, mix_mono=False,
+    # package_name='infinite_music_discs', dest_dir='.', compress_output=False)
+
+    def __init__(self):
+        parser = ArgumentParser(prog="mc music packer")
+
+        parser.add_argument("-f", "--disc-entries-file", type=str, required=True,
+                            help="a csv file delimited by semicolon, "
+                                 "each line of which contains texture_file;track_file;title;internal_name.")
+        parser.add_argument("-g", "--minecraft-version", type=str, required=True,
+                            help="specify a game version, support 1.14 - 1.20.2",
+                            default="1.20.2")
+        parser.add_argument("-p", "--use-parallel", action='store_true',
+                            help="allow convert operations run in parallel")
+        parser.add_argument("-m", "--mix-mono", action='store_true',
+                            help="mix mono, just like the song is played from the jukebox.")
+        parser.add_argument("-n", "--package-name", type=str, default="infinite_music_discs",
+                            help="package name")
+        parser.add_argument("-o", "--output-dir", type=str, default=".", help="the output directory of rp&dp.")
+        parser.add_argument("-z", "--compress-output", action='store_true', help="compress output pack to zip")
+        parser.add_argument("-c", "--cover-image", type=str, default='', help="cover image of the pack")
```

Comment:
Fix: the CLI throws an exception if you don't set the `cover-image` flag. Are you changing the value somewhere between here and when it's used in the generator? Because if the pack.png path is set to an empty string like you've done here, it should work without throwing an exception.